### DISCUSSION TITLE
How to use AirPlay in PlatformIO C++ code?

### DISCUSSION THREAD

**Discussion by hientv1999** [sle118/squeezelite-esp32#444] [Q&A]
I'm building my speaker. It uses vastly different customized hardware and it has beat-synced LED, EQ control, and other GPIO settings that I don't want to use SqueezeLite. However, I want to add AirPlay to my project. I looked through the project code and it seems to be written for ESP IDF. I believe the AirPlay code is in raop and wifi_manger folders. Did I miss any other file? Has anyone done the same that can share your source code? It will be best if the code is written for PlatformIO arduino framework

**Comment by philippe44** [comment]
It should be all under `raop`. The rest is porting (I initially use my libraop library and had a dedicated test version under Windows, you still can see some WIN32 defined). Good luck!
